### PR TITLE
Use a script to bump a version and push to master

### PR DESCRIPTION
**Pull request by lumberj**
Adds `npm run bump` which will:

1) Check we're on master
2) Check local is in sync with remote (not diverged, ahead or behind)
2a) Print a list of the commits since the last released version
3) Bump version locally, commit and tag
4) Push commit and tags to remote

### CODE REVIEW

**Comment by lumberj** [comment]
cc @geertweening

**Comment by geertweening** [comment]
This looks good, +1 would like someone else to review as well

**Comment by lumberj** [comment]
@sentientwaffle reviewed this for correctness, and I also used [shellcheck](www.shellcheck.net)

File: bin/bump.sh [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+      npm version major
+      ;;
+    minor)
+      npm version minor
+      ;;
+    patch)
+      npm version patch
+      ;;
+    *)
+      echo "Usage npm run bump {major|minor|patch}"
```
Comment (clark800) [inline bin/bump.sh]: Did you test this usage? The npm docs say that you are supposed to use `--`: https://docs.npmjs.com/cli/run-script

Comment (lumberj) [inline bin/bump.sh]: Yes, works as expected

**Pull request merged**